File: csharp/Structural/08-Composite/Leaf/JobCard.cs
using Composite.Component;

namespace Composite.Leaf;

public class JobCard : IJobComponent
{
    public string Name { get; }
    public decimal Cost { get; }
    public string Category { get; }
    public string Assignee { get; set; } = "Unassigned";

    public JobCard(string name, string category, decimal cost)
    {
        Name = name;
        Category = category;
        Cost = cost;
    }

    public void Execute() =>
        Console.WriteLine($"{"".PadLeft(4)}✅ Executing: [{Category}] {Name} — N${Cost:N2} | Assigned to: {Assignee}");

    public void Display(int depth = 0) =>
        Console.WriteLine($"{"".PadLeft(depth * 2)}📋 {Name} [{Category}] — N${Cost:N2}");
}

File: typescript/Structural/08-Composite/Leaf/JobCard.ts
import { IJobComponent } from "../Component/IJobComponent";

export class JobCard implements IJobComponent {
  assignee: string = "Unassigned";

  constructor(
    readonly name: string,
    readonly category: string,
    readonly cost: number,
  ) {}

  execute(): void {
    console.log(
      `${"  ".repeat(2)}✅ Executing: [${this.category}] ${this.name} — N$${this.cost.toFixed(2)} | Assigned to: ${this.assignee}`,
    );
  }

  display(depth: number = 0): void {
    console.log(
      `${"  ".repeat(depth)}📋 ${this.name} [${this.category}] — N$${this.cost.toFixed(2)}`,
    );
  }
}
